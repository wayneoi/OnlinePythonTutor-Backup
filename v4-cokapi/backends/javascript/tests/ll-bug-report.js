// https://github.com/pgbovine/OnlinePythonTutor/issues/244
// bug: execution terminates after 2 steps
class LinkedListNode {
    constructor(value, next = null) {
        this.value = value;
        this.next = next;
    }

    toString() {
        return `->${this.value.age}`;
    }

}
const F = Symbol("femmme");
const H = Symbol("homme");
const listPers = new LinkedListNode({name: "gag",age: 12,sex: F});

console.log(listPers.toString());
