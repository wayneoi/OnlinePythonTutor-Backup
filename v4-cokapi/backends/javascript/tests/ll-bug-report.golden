{ code: '// https://github.com/pgbovine/OnlinePythonTutor/issues/244\n// bug: execution terminates after 2 steps\nclass LinkedListNode {\n    constructor(value, next = null) {\n        this.value = value;\n        this.next = next;\n    }\n\n    toString() {\n        return `->${this.value.age}`;\n    }\n\n}\nconst F = Symbol("femmme");\nconst H = Symbol("homme");\nconst listPers = new LinkedListNode({name: "gag",age: 12,sex: F});\n\nconsole.log(listPers.toString());',
  trace: 
   [ { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { LinkedListNode: [ 'REF', 1 ] },
       ordered_globals: [ 'LinkedListNode' ],
       line: 14,
       col: 0,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'LinkedListNode',
             'class LinkedListNode {\n    constructor(value, next = null) {\n        this.value = value;\n        this.next = next;\n    }\n\n    toString() {\n        return `->${this.value.age}`;\n    }\n\n}',
             null,
             null ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: 
        { LinkedListNode: [ 'REF', 1 ],
          F: [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ] },
       ordered_globals: [ 'LinkedListNode', 'F' ],
       line: 15,
       col: 0,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'LinkedListNode',
             'class LinkedListNode {\n    constructor(value, next = null) {\n        this.value = value;\n        this.next = next;\n    }\n\n    toString() {\n        return `->${this.value.age}`;\n    }\n\n}',
             null,
             null ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: 
        { LinkedListNode: [ 'REF', 1 ],
          F: [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ],
          H: [ 'JS_SPECIAL_VAL', 'Symbol(homme)' ] },
       ordered_globals: [ 'LinkedListNode', 'F', 'H' ],
       line: 16,
       col: 0,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'LinkedListNode',
             'class LinkedListNode {\n    constructor(value, next = null) {\n        this.value = value;\n        this.next = next;\n    }\n\n    toString() {\n        return `->${this.value.age}`;\n    }\n\n}',
             null,
             null ] } },
     { stdout: '',
       func_name: 'LinkedListNode',
       stack_to_render: 
        [ { func_name: 'LinkedListNode (constructor)',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'LinkedListNode (constructor)_f2',
            ordered_varnames: [ 'this', 'value', 'next' ],
            encoded_locals: 
             { this: [ 'REF', 2 ],
               value: [ 'REF', 3 ],
               next: [ 'JS_SPECIAL_VAL', 'null' ] } } ],
       globals: 
        { LinkedListNode: [ 'REF', 1 ],
          F: [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ],
          H: [ 'JS_SPECIAL_VAL', 'Symbol(homme)' ] },
       ordered_globals: [ 'LinkedListNode', 'F', 'H' ],
       line: 5,
       col: 8,
       event: 'call',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'LinkedListNode',
             'class LinkedListNode {\n    constructor(value, next = null) {\n        this.value = value;\n        this.next = next;\n    }\n\n    toString() {\n        return `->${this.value.age}`;\n    }\n\n}',
             null,
             null ],
          '2': [ 'INSTANCE', '' ],
          '3': 
           [ 'INSTANCE',
             '',
             [ 'name', 'gag' ],
             [ 'age', 12 ],
             [ 'sex', [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ] ] ] } },
     { stdout: '',
       func_name: 'LinkedListNode',
       stack_to_render: 
        [ { func_name: 'LinkedListNode (constructor)',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'LinkedListNode (constructor)_f2',
            ordered_varnames: [ 'this', 'value', 'next' ],
            encoded_locals: 
             { this: [ 'REF', 2 ],
               value: [ 'REF', 3 ],
               next: [ 'JS_SPECIAL_VAL', 'null' ] } } ],
       globals: 
        { LinkedListNode: [ 'REF', 1 ],
          F: [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ],
          H: [ 'JS_SPECIAL_VAL', 'Symbol(homme)' ] },
       ordered_globals: [ 'LinkedListNode', 'F', 'H' ],
       line: 6,
       col: 8,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'LinkedListNode',
             'class LinkedListNode {\n    constructor(value, next = null) {\n        this.value = value;\n        this.next = next;\n    }\n\n    toString() {\n        return `->${this.value.age}`;\n    }\n\n}',
             null,
             null ],
          '2': [ 'INSTANCE', '', [ 'value', [ 'REF', 3 ] ] ],
          '3': 
           [ 'INSTANCE',
             '',
             [ 'name', 'gag' ],
             [ 'age', 12 ],
             [ 'sex', [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ] ] ] } },
     { stdout: '',
       func_name: 'LinkedListNode',
       stack_to_render: 
        [ { func_name: 'LinkedListNode (constructor)',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'LinkedListNode (constructor)_f2',
            ordered_varnames: [ 'this', 'value', 'next', '__return__' ],
            encoded_locals: 
             { this: [ 'REF', 2 ],
               value: [ 'REF', 3 ],
               next: [ 'JS_SPECIAL_VAL', 'null' ],
               __return__: [ 'REF', 2 ] } } ],
       globals: 
        { LinkedListNode: [ 'REF', 1 ],
          F: [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ],
          H: [ 'JS_SPECIAL_VAL', 'Symbol(homme)' ] },
       ordered_globals: [ 'LinkedListNode', 'F', 'H' ],
       line: 6,
       col: 4,
       event: 'return',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'LinkedListNode',
             'class LinkedListNode {\n    constructor(value, next = null) {\n        this.value = value;\n        this.next = next;\n    }\n\n    toString() {\n        return `->${this.value.age}`;\n    }\n\n}',
             null,
             null ],
          '2': 
           [ 'INSTANCE',
             '',
             [ 'value', [ 'REF', 3 ] ],
             [ 'next', [ 'JS_SPECIAL_VAL', 'null' ] ] ],
          '3': 
           [ 'INSTANCE',
             '',
             [ 'name', 'gag' ],
             [ 'age', 12 ],
             [ 'sex', [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ] ] ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: 
        { LinkedListNode: [ 'REF', 1 ],
          F: [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ],
          H: [ 'JS_SPECIAL_VAL', 'Symbol(homme)' ],
          listPers: [ 'REF', 2 ] },
       ordered_globals: [ 'LinkedListNode', 'F', 'H', 'listPers' ],
       line: 18,
       col: 0,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'LinkedListNode',
             'class LinkedListNode {\n    constructor(value, next = null) {\n        this.value = value;\n        this.next = next;\n    }\n\n    toString() {\n        return `->${this.value.age}`;\n    }\n\n}',
             null,
             null ],
          '2': 
           [ 'INSTANCE',
             '',
             [ 'value', [ 'REF', 3 ] ],
             [ 'next', [ 'JS_SPECIAL_VAL', 'null' ] ] ],
          '3': 
           [ 'INSTANCE',
             '',
             [ 'name', 'gag' ],
             [ 'age', 12 ],
             [ 'sex', [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ] ] ] } },
     { stdout: '',
       func_name: 'toString',
       stack_to_render: 
        [ { func_name: 'toString',
            frame_id: 3,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'toString_f3',
            ordered_varnames: [ 'this' ],
            encoded_locals: { this: [ 'REF', 2 ] } } ],
       globals: 
        { LinkedListNode: [ 'REF', 1 ],
          F: [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ],
          H: [ 'JS_SPECIAL_VAL', 'Symbol(homme)' ],
          listPers: [ 'REF', 2 ] },
       ordered_globals: [ 'LinkedListNode', 'F', 'H', 'listPers' ],
       line: 10,
       col: 8,
       event: 'call',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'LinkedListNode',
             'class LinkedListNode {\n    constructor(value, next = null) {\n        this.value = value;\n        this.next = next;\n    }\n\n    toString() {\n        return `->${this.value.age}`;\n    }\n\n}',
             null,
             null ],
          '2': 
           [ 'INSTANCE',
             '',
             [ 'value', [ 'REF', 3 ] ],
             [ 'next', [ 'JS_SPECIAL_VAL', 'null' ] ] ],
          '3': 
           [ 'INSTANCE',
             '',
             [ 'name', 'gag' ],
             [ 'age', 12 ],
             [ 'sex', [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ] ] ] } },
     { stdout: '',
       func_name: 'toString',
       stack_to_render: 
        [ { func_name: 'toString',
            frame_id: 3,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'toString_f3',
            ordered_varnames: [ 'this', '__return__' ],
            encoded_locals: { this: [ 'REF', 2 ], __return__: '->12' } } ],
       globals: 
        { LinkedListNode: [ 'REF', 1 ],
          F: [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ],
          H: [ 'JS_SPECIAL_VAL', 'Symbol(homme)' ],
          listPers: [ 'REF', 2 ] },
       ordered_globals: [ 'LinkedListNode', 'F', 'H', 'listPers' ],
       line: 10,
       col: 4,
       event: 'return',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'LinkedListNode',
             'class LinkedListNode {\n    constructor(value, next = null) {\n        this.value = value;\n        this.next = next;\n    }\n\n    toString() {\n        return `->${this.value.age}`;\n    }\n\n}',
             null,
             null ],
          '2': 
           [ 'INSTANCE',
             '',
             [ 'value', [ 'REF', 3 ] ],
             [ 'next', [ 'JS_SPECIAL_VAL', 'null' ] ] ],
          '3': 
           [ 'INSTANCE',
             '',
             [ 'name', 'gag' ],
             [ 'age', 12 ],
             [ 'sex', [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ] ] ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: 
        { LinkedListNode: [ 'REF', 1 ],
          F: [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ],
          H: [ 'JS_SPECIAL_VAL', 'Symbol(homme)' ],
          listPers: [ 'REF', 2 ] },
       ordered_globals: [ 'LinkedListNode', 'F', 'H', 'listPers' ],
       line: 18,
       col: 8,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'LinkedListNode',
             'class LinkedListNode {\n    constructor(value, next = null) {\n        this.value = value;\n        this.next = next;\n    }\n\n    toString() {\n        return `->${this.value.age}`;\n    }\n\n}',
             null,
             null ],
          '2': 
           [ 'INSTANCE',
             '',
             [ 'value', [ 'REF', 3 ] ],
             [ 'next', [ 'JS_SPECIAL_VAL', 'null' ] ] ],
          '3': 
           [ 'INSTANCE',
             '',
             [ 'name', 'gag' ],
             [ 'age', 12 ],
             [ 'sex', [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ] ] ] } },
     { stdout: '->12\n',
       func_name: '',
       stack_to_render: [],
       globals: 
        { LinkedListNode: [ 'REF', 1 ],
          F: [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ],
          H: [ 'JS_SPECIAL_VAL', 'Symbol(homme)' ],
          listPers: [ 'REF', 2 ] },
       ordered_globals: [ 'LinkedListNode', 'F', 'H', 'listPers' ],
       line: 18,
       col: 33,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'LinkedListNode',
             'class LinkedListNode {\n    constructor(value, next = null) {\n        this.value = value;\n        this.next = next;\n    }\n\n    toString() {\n        return `->${this.value.age}`;\n    }\n\n}',
             null,
             null ],
          '2': 
           [ 'INSTANCE',
             '',
             [ 'value', [ 'REF', 3 ] ],
             [ 'next', [ 'JS_SPECIAL_VAL', 'null' ] ] ],
          '3': 
           [ 'INSTANCE',
             '',
             [ 'name', 'gag' ],
             [ 'age', 12 ],
             [ 'sex', [ 'JS_SPECIAL_VAL', 'Symbol(femmme)' ] ] ] } } ] }
